### ISSUE TITLE
Add support for JPEG XL

### ISSUE THREAD

**Issue by duongdominhchau** [arsenetar/dupeguru#1109]
First, thank you for building this awesome tool, it helped a lot in maintaining my image vault.

Now, back to the problem: JPEG XL takes less space to store the same image, so I converted all my images to that format. However, after the conversion, I realized dupeguru does not have the ability to read that format yet, so now no duplicated image detection anymore. Therefore, here's an issue asking for JPEG XL support.

I use libjxl 0.8.1 on ArchLinux to convert the images. For now I don't know if there is any other information needed, if I miss anything please let me know so I can fill in, thanks.

**Comment by glubsy** [comment]
Definitely something to look into.

**Comment by IlluminatiWave** [comment]
https://github.com/easymodo/qimgv

Qimgv Supports multiple file formats and Qt5. I tried to copy Qimgv libraries to Dupeguru and got no results. (includes AVIF, HEIC, APNG, JP2, JpegXL, MNG and RAW)

**Comment by chikari802** [comment]
Support for JPEG XL in Qt already exists on Linux. If `kimageformat-plugins` version 5.89 or newer is installed (or alternatively if `qt-jpegxl-image-plugin` is installed), simply patching `core/pe/photo.py` by adding `jxl` to `HANDLED_EXTS` is enough to make it work.

**Comment by glubsy** [comment]
> simply patching `core/pe/photo.py` by adding `jxl` to `HANDLED_EXTS` is enough to make it work.

have you actually tested this? I'm a bit skeptical, because last time someone mentioned something similar to support the webp format, while it may have enabled Dupeguru to read the webp files, but it was not enough to actually detect dupes (from my experience).

**Comment by chikari802** [comment]
@glubsy yes, I have tested this. It correctly detects 100% dupes and the rest (97-99%) are pretty close, not perfect but usable enough for me. Also the preview window works perfectly. (The only thing not working at all is animations in the preview but that seems to be unrelated since GIFs do not animate either for me)

**Comment by mtp-lac** [comment]
Bumping it as JXL support is coming to browsers next month. I really hope there will be official support for reading image contents now
